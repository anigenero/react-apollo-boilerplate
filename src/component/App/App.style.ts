import { Theme } from '@material-ui/core';
import { createStyles, StyleRules } from '@material-ui/styles';

export type AppCssKey =
    'grow' |
    'header' |
    'title';

export const appStyles =
    (theme: Theme): StyleRules<{}, AppCssKey> =>
        createStyles({
            grow: {
                flex: 1
            },
            header: {
                boxShadow: theme.shadows[2]
            },
            title: {
                fontWeight: 'bold'
            },
        });
