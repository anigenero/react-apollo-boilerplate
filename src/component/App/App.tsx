import { Typography } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Toolbar from '@material-ui/core/Toolbar';
import EditIcon from '@material-ui/icons/Edit';
import * as React from 'react';
import { FunctionComponent, useEffect, useState } from 'react';
import { LocalizeContextProps, Translate, withLocalize } from 'react-localize-redux';
import { AppLocale } from '../../assets/locale/generated';
import { TaskList } from '../TaskList/TaskList';
import { appStyles } from './App.style';

const englishTranslations = require('../../assets/locale/en-US.json');

const useClasses = makeStyles(appStyles);

export const App: FunctionComponent<LocalizeContextProps> = ({initialize, addTranslationForLanguage}) => {

    const classes = useClasses({});

    const [edit, setEdit] = useState(false);

    useEffect(() => {

        initialize({
            languages: [{
                code: 'en-US',
                name: 'English (US)'
            }],
            options: {
                defaultLanguage: 'en-US',
                renderInnerHtml: false,
                renderToStaticMarkup: false
            }
        });

        addTranslationForLanguage(englishTranslations, 'en-US');

    }, []);

    return (
        <>

            <AppBar position="sticky" className={classes.header}>
                <Toolbar>

                    <Typography variant="h6" color="secondary" className={classes.title}>
                        <Translate id={AppLocale.app.longName}/>
                    </Typography>

                    <div className={classes.grow}/>

                    <IconButton color={edit ? 'secondary' : 'default'} onClick={() => setEdit(!edit)}>
                        <EditIcon/>
                    </IconButton>

                </Toolbar>
            </AppBar>

            <TaskList edit={edit}/>

        </>
    );

};

export default withLocalize(App);
