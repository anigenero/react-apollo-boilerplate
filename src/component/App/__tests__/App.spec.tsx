import React from 'react';
import { ValidatorProvider } from 'react-class-validator';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import configureStore, { MockStoreEnhanced } from 'redux-mock-store';
import { AppState } from '../../../redux/app.state';
import App from '../App';

const mockStore = configureStore<Partial<AppState>>([]);

describe('App', () => {

    let store: MockStoreEnhanced<Partial<AppState>>;

    beforeEach(() => {
        store = mockStore({
            task: {
                tasks: []
            },
        });
    });

    test('should render correctly', () => {

        const tree = create(
            <Provider store={store}>
                <ValidatorProvider>
                    <App/>
                </ValidatorProvider>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

});
