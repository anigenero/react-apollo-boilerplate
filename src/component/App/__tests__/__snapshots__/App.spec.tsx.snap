// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-header-2 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-colorSecondary"
      >
        
      </h6>
      <div
        className="makeStyles-grow-1"
      />
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>,
  <ul
    className="MuiList-root MuiList-padding"
  />,
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <form
      onSubmit={[Function]}
      style={
        Object {
          "display": "flex",
          "flex": 1,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-control-4"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
        >
          
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
            â€‰
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            name="task"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          
        </span>
      </button>
    </form>
  </div>,
]
`;
